SharedSource/Core/FunctionalProgramming/EnumerableExtensions.cs

```diff
@@ -190,12 +191,12 @@ public static IEnumerable<TSource> CreateSequence<TSource> (this TSource source,
     /// A sequence of elements containing the <paramref name="source"/> and all subsequent elements 
     /// until the <paramref name="nextElementSelector"/> returns <see langword="null" />.
     /// </returns>
-    public static IEnumerable<TSource> CreateSequence<TSource> (this TSource source, Func<TSource, TSource> nextElementSelector)
+    public static IEnumerable<TSource> CreateSequence<TSource> (this TSource source, Func<TSource, TSource?> nextElementSelector)
         where TSource : class
     {
       ArgumentUtility.CheckNotNull ("nextElementSelector", nextElementSelector);
 
-      return CreateSequence (source, nextElementSelector, e => e != null);
+      return CreateSequence (source, nextElementSelector!, e => e != null)!;

```

Comment:
interesting. this pr does not already contain the null-detection update for enumerable extensions, correct?

Reply:
It does now.